bears/documentation/DocumentationStyleBear.py

```diff
@@ -0,0 +1,118 @@
+                following lines are also 4 space indented
+
+        :param language: The programming language of the file(s).
+        :param docstyle: The docstyle to use. For example ``default`` or
+                         ``doxygen``. Docstyles are language dependent, meaning
+                         not every language is supported by a certain docstyle.
+        :param allow_missing_func_desc: When set ``True`` this will allow
+                         functions with missing descriptions, allowing
+                         functions to start with params.
+        """
+        for doc_comment in extract_documentation(file, language, docstyle):
+            parsed = doc_comment.parse()
+            metadata = iter(parsed)
+
+            # Assuming that the first element is always the only main
+            # description.
+            main_description = next(metadata)
+
+            if main_description.desc == '\n' and not allow_missing_func_desc:
+                warning_desc = """
+Missing function description.
+Please set allow_missing_func_desc = True to ignore this warning.
+"""
+            else:
+                warning_desc = 'Documentation does not have correct style.'
+
+            # one empty line shall follow main description (except it's empty
+            # or no annotations follow).
+            if main_description.desc.strip() != '':
+                main_description = main_description._replace(
+                    desc='\n' + main_description.desc.strip() + '\n' *
+                         (1 if len(parsed) == 1 else 2))
+
+            new_metadata = [main_description]
+            for m in metadata:
+                # Split newlines and remove leading and trailing whitespaces.
+                stripped_desc = list(map(str.strip, m.desc.splitlines()))
+
+                if len(stripped_desc) == 0:
+                    # Annotations should be on their own line, though no
+                    # further description follows.
+                    stripped_desc.append('')
+                else:
+                    # Wrap parameter description onto next line if it follows
+                    # annotation directly.
+                    if stripped_desc[0] != '':
+                        stripped_desc.insert(0, '')
+
+                # Indent with 4 spaces.
+                stripped_desc = ('' if line == '' else ' ' * 4 + line for line
+                                 in stripped_desc)
+
+                new_desc = '\n'.join(stripped_desc)
+
+                # Strip away trailing whitespaces and obsolete newlines (except
+                # one newline which is mandatory).
+                new_desc = new_desc.rstrip() + '\n'
+
+                new_metadata.append(m._replace(desc=new_desc.lstrip(' ')))
+
+            new_comment = DocumentationComment.from_metadata(
+                new_metadata, doc_comment.docstyle_definition,
+                doc_comment.marker, doc_comment.indent, doc_comment.range)
```

Comment:
indent size is indent size!

`here 4 spaces in param desc` is indented.  It is indented by the same indent size that is used everywhere else.

yes, there are crazy people who will want different parts of their code indented at magically different levels, but that is unlikely to be something coala will ever support.  What coala does support is one `indent_size`, chosen by the user , which is the size that should be used everywhere an indent is expected.  If a user asks for `indent_size = 2`, that should be the indent size *everywhere* in the selected files.  Anywhere it doesnt work is a bug.

We also support `use_spaces = False`, which I believe also doesnt work, and so far this PR doesnt have any tests for, but I have given up on that.